src/basic/fileio.c

```diff
@@ -375,9 +398,8 @@ int verify_file(const char *fn, const char *blob, bool accept_extra_nl) {
         return 1;
 }
 
-int read_virtual_file(const char *filename, size_t max_size, char **ret_contents, size_t *ret_size) {
+int read_virtual_stream(int fd, size_t max_size, char **ret_contents, size_t *ret_size) {
```

Comment:
call this: read_virtual_file_fd() please

("stream" is what FILE* is called, but we don#t have that here, we have a low-level fd)